torch/nn/init.py

```diff
@@ -365,12 +365,10 @@ def sparse_(tensor, sparsity, std=0.01):
     with torch.no_grad():
         tensor.normal_(0, std)
         for col_idx in range(cols):
-            row_indices = list(range(rows))
-            random.shuffle(row_indices)
+            row_indices = torch.randperm(rows)
```

Comment:
nice! I did a quick micro-benchmark and it looks like using `randperm` is much faster than creating the list in python and then doing `random.shuffle`!